railties/lib/rails/commands/server/server_command.rb

```diff
@@ -256,6 +259,12 @@ def early_hints
           options[:early_hints]
         end
 
+        def log_to_stdout?
+          return options[:log_to_stdout] unless options[:log_to_stdout].nil?
+
+          options[:daemon].blank? && environment == "development"
```

Comment:
Let's go with:

```ruby
def log_to_stdout?
  options.fetch(:log_to_stdout) do
    options[:daemon].blank? && environment == "development"
  end
end
```

That, coupled with the `optional: true, default: nil` in the `class_option` should make it such that `--log-to-stdout` or `--no-log-to-stdout` always overrides.